compiler/src/dotty/tools/dotc/FromTasty.scala

```diff
@@ -71,12 +71,14 @@ object FromTasty extends Driver {
       case unit: TASTYCompilationUnit =>
         assert(ctx.settings.YretainTrees.value)
         val className = unit.className.toTypeName
-        val compilationUnits = List(tree(className), tree(className.moduleClassName)).flatMap {
-          case Some((clsd, unpickled)) if !unpickled.isEmpty =>
-            List(CompilationUnit.mkCompilationUnit(clsd, unpickled, forceTrees = true))
-          case _ => Nil
+        def compilationUnit(className: TypeName): Option[CompilationUnit] = {
+          tree(className).flatMap { case (clsd, unpickled) =>
+            if (unpickled.isEmpty) None
+            else Some(CompilationUnit.mkCompilationUnit(clsd, unpickled, forceTrees = true))
+          }
         }
-        compilationUnits
+        // Try to load it from the class, or else if the class is not present try to load it from the object with that name
+        compilationUnit(className).orElse(compilationUnit(className.moduleClassName)).toList
```

Comment:
Return type should be an Option and not a List now I guess.

Reply:
Done. It was only IntelliJ that complained about the `Option` in the flat map.